### ISSUE TITLE
please include documentation to *not* set the container hostname

### ISSUE THREAD

**Issue by bnfinet** [JrCs/docker-letsencrypt-nginx-proxy-companion#162]
when the container is run with a hostname such as

```
docker run -d \
    -h my-host-dockergen-lets-encrypt \
    -v /path/to/certs:/etc/nginx/certs:rw \
    --volumes-from nginx-proxy \
    -v /var/run/docker.sock:/var/run/docker.sock:ro \
    jrcs/letsencrypt-nginx-proxy-companion
```

then this resulting curl call from within the `docker_api` function fails...
`curl -s --unix-socket /var/run/docker.sock -XGET http://localhost/containers/my-host-dockergen-lets-encrypt/json`

https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion/blob/master/app/functions.sh#L51

I'd be happy to provide a small PR if you're favorable to this inclusion.

**Comment by JrCs** [comment]
Sorry, i don't understand why the function fails if you are giving a hostname to your container ?
The docker API use container ID not hostname.

**Comment by bnfinet** [comment]
this is the call to `docker_api` from within `check_writable_directory` which fails due to it's use of `$HOSTNAME`:

`docker_api "/containers/$HOSTNAME/json" | jq ".Mounts[].Destination" | grep -q "^\"$dir\"$"`

https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion/blob/master/app/entrypoint.sh#L45

**Comment by JrCs** [comment]
Commit 8b97f203
@bnfinet can you check with the dev version of the container ?

**Comment by joshontheweb** [comment]
@JrCs I had this same issue due to the fact that I'm deploying via Docker Cloud and they set a hostname for you by default.  I tested this with the :dev version and it has fixed the issue.  Thank you!  I can move forward with :dev for now but a tagged container with this change would be welcome :)

**Comment by buchdag** [comment]
This was fixed on master by commit 86ee879
